---
bool is_move_complete
